frontend/src/views/ProjectView.tsx

```diff
@@ -1,22 +1,72 @@
-import ProjectList from "../components/ProjectList";
-import "../styles/ProjectView.css";
-import "../styles/ProjectList.css";
-import CreateProjectDialog from "../components/CreateProjectDialog";
-import { Token } from "../contexts/Auth.tsx";
+import { useState } from 'react';
+import '../styles/CreateProjectDialog.css';
+import { useParams } from 'react-router';
+import Hardware from '../components/HardwareSetClass';
+import call, { Method } from '../utils/api';
+import HardwareDialog from '../components/HardwareDialog';
 
-interface ProjectViewProps {
-    token: Token;
+// Define an enum for error types
+enum ErrorType {
+    None = '',
+    PageOpenError = 'Failed to open project-view. Please try again.',
+    HardwareError ='No hardware set found.'
 }
 
-export default function ProjectView({ token }: Readonly<ProjectViewProps>) {
+
+const ProjectView = () => {
```

Comment:
Try to keep code style consistent, so in this case all of our components use the style `export default function Component(...)`